frontends/p4/tableKeyNames.h

```diff
@@ -38,6 +31,24 @@ class DoTableKeyNames : public Transform {
     const IR::Node* postorder(IR::KeyElement* keyElement) override;
 };
 
+/** Adds a "@name" annotation to each table key that does not have a name.
+ * The string used for the name is derived from the expression itself - if
+ * the expression is "simple" enough.  If the expression is not simple the
+ * compiler will give an error.  Simple expressions are:
+ * - .isValid(),
+ * - ArrayIndex,
+ * - BAnd (where at least one operand is constant),
```

Comment:
I think that the second operand must be constant.
And I think that this case does not belong here.